designs/2020-init-command-eslint-cli/README.md

```diff
@@ -26,17 +25,45 @@ We will make use of tools like `npx` to simply run `npx @eslint/create` or using
    used. Be sure to define any new terms in this section.
 -->
 
-The implementation is mainly migrating the code from main repo to `eslint-cli`. As far as the modules are concerned, all shared modules are being shipped currently with the eslint so we can use those modules through eslint itself. 
-As far as the `eslint-recommended` rules are concerned, it will be done as a copy paste to the `@eslint/create` repo. 
+The implementation is mainly migrating the code from main repo to `@eslint/create`. As far as the modules are concerned, all shared modules are being shipped currently with the eslint so we can use those modules through eslint itself. 
+
+The approach would be following these steps
+
+- Move the `eslint/lib/init/*` to the new repo's `src/*` directory
+- Add the following `dependencies` in `package.json`
+  - `enquirer`
+  - `progress`
+  - `semver`
+  - `espree` (peer)
+  - `lodash`
+  - `json-stable-stringify-without-jsonify`
+  - `cross-spawn`
+  - `eslint` (peer)
+  - `debug` (peer)
+- for all those modules coming inside `../**/*`, it would be replaced using `eslint/**/*`
+- For `tests`, move the `tests/lib/init` to new repo's `tests/init`
+- Add the following `devDependencies` in `package.json`
+  - `chai`
+  - `sinon`
+  - `js-yaml`
+  - `proxyquire`
+  - `shelljs`
+- create `tests/utils` and add the following functions 
+  - [`defineInMemoryFs`](https://github.com/eslint/eslint/blob/6677180495e16a02d150d0552e7e5d5f6b77fcc5/tests/_utils/in-memory-fs.js#L250)
+- for `conf/*`, it is not a part of the public API and it cant be accessed from `eslint` module, we need a keep these synced with the main repo (`eslint`). We can choose from the following approach to solve this : 
+  - using the `eslint-github-bot`, whenever there is a PR opened that changes the file for the `conf` directory in `eslint` repo, create an issue in the `@eslint/create` repo to have a look in the PR and if required submit a change in `@eslint/create`.
+  - OR, while doing the release for both `eslint` and `@eslint/create` repos, the release script can be configured to copy-parse the `conf` folder from `eslint` to `@eslint/create` before the release just like the `website` repo is being updated currently. 
+- After completion of this repo, whenever someone uses the `--init` flag, an error will be thrown stating the correct step to do the migration i.e using `npx @eslint/create` or `npm @eslint` or `yarn @eslint`. I am not suggesting to use run the `@eslint/create` module whenever `--init` command is being called because this would require to use the `@eslint/create` module as a dependency in `eslint` and in `@eslint/create` we are already being using `eslint` as a dependency, so there would be circular dependencies.
```

Comment:
Why does `@eslint/create` need to depend directly on `eslint`?

Reply:
I’d say let’s just get rid of auto config altogether. I don’t think it’s that useful anymore. Plus, we can avoid a direct dependency on ESLint that way.

Reply:
A tool like [eslint-nibble](https://github.com/IanVS/eslint-nibble) will be a much better experience than autoconfig, so I have no problem with removing it :+1:

Reply:
Sorry but I am not getting what do you mean with auto-config ? is it the whole --init all together? or `conf/*` ? or something else ?

Reply:
If you go through `--init` now, you get to a point where it asks if you want us to examine the source code files to determine the correct config to use. That's called autoconfig. Most of the functionality is in `lib/init/autoconfig.js`, so it would just be a matter of removing that file and then removing the associated question that triggers it.

Reply:
Cool, Thanks for the details. 
 make sense to remove autoconfig

But still, we need to use `eslint` as (peer)-dependency as mentioned [here](https://github.com/eslint/rfcs/pull/58#discussion_r475226750)

Reply:
Are linter and CLIEngine used outside of auto config?

Reply:
no I guess, cliEngine is being used in `source-code-utils` but I guess if we are removing the autoconfig then we don't need that either.

Reply:
Okay, sounds like we are on the right track. I'd suggest updating this RFC with the latest agreements. I think we're close!

Reply:
I had such high hopes for autoconfig, but I agree that it was never terribly useful, partly because it only worked if you were already strictly following a convention.  I'm actually surprised it hasn't been removed already.  :)

This RFC looks like a great move. I love what y'all are doing and hope everyone is well and having a happy new year!